apps/full-site-editing/full-site-editing-plugin/donations/donations.js

```diff
@@ -6,53 +6,106 @@ import classNames from 'classnames';
 /**
  * WordPress dependencies
  */
-import { useState } from '@wordpress/element';
+import { InspectorControls, __experimentalBlock as Block } from '@wordpress/block-editor';
+import { Button, ExternalLink, PanelBody, ToggleControl } from '@wordpress/components';
+import { useEffect, useState } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
-import { Button } from '@wordpress/components';
 
 /**
  * Internal dependencies
  */
 import StripeNudge from './stripe-nudge';
 
-const Donations = ( { products, stripeConnectUrl } ) => {
+const Donations = ( { attributes, products, setAttributes, siteSlug, stripeConnectUrl } ) => {
+	const { oneTimePlanId, monthlyPlanId, annuallyPlanId, showCustomAmount } = attributes;
 	const [ activeTab, setActiveTab ] = useState( 'one-time' );
 
-	const isActive = ( button ) => ( activeTab === button ? 'active' : null );
+	const isTabActive = ( tab ) => activeTab === tab;
+
+	useEffect( () => {
+		if ( ! oneTimePlanId ) {
+			setAttributes( { oneTimePlanId: products[ 'one-time' ] } );
```

Comment:
Let's start with this but I'm not sure if we want to start with one interval type, or all 3.